pay-api/src/pay_api/services/eft_gl_transfer.py

```diff
@@ -76,33 +76,27 @@ def search(search_criteria: EFTGlTransferSearch = EFTGlTransferSearch()) -> [EFT
         """Return EFT Transfers by search criteria."""
         query = db.session.query(EFTGLTransferModel)
 
-        if search_criteria.created_on:
-            query = query.filter(func.DATE(EFTGLTransferModel.created_on) == search_criteria.created_on.date())
+        query = EFTGlTransfer.filterIfPresent(query, search_criteria.created_on, EFTGLTransferModel.created_on)
+        query = EFTGlTransfer.filterIfPresent(query, search_criteria.invoice_id, EFTGLTransferModel.invoice_id)
+        query = EFTGlTransfer.filterIfPresent(query, search_criteria.is_processed, EFTGLTransferModel.is_processed)
+        query = EFTGlTransfer.filterIfPresent(query, search_criteria.processed_on, EFTGLTransferModel.processed_on)
+        query = EFTGlTransfer.filterIfPresent(query, search_criteria.short_name_id, EFTGLTransferModel.short_name_id)
+        query = EFTGlTransfer.filterIfPresent(query, search_criteria.source_gl, EFTGLTransferModel.source_gl)
+        query = EFTGlTransfer.filterIfPresent(query, search_criteria.target_gl, EFTGLTransferModel.target_gl)
+        query = EFTGlTransfer.filterIfPresent(query, search_criteria.transfer_type, EFTGLTransferModel.transfer_type)
+        query = EFTGlTransfer.filterIfPresent(query, search_criteria.transfer_date, EFTGLTransferModel.transfer_date)
 
-        if search_criteria.invoice_id:
-            query = query.filter(EFTGLTransferModel.invoice_id == search_criteria.invoice_id)
-
-        if search_criteria.is_processed is not None:
-            query = query.filter(EFTGLTransferModel.is_processed == search_criteria.is_processed)
-
-        if search_criteria.processed_on:
-            query = query.filter(func.DATE(EFTGLTransferModel.processed_on) == search_criteria.processed_on.date())
-
-        if search_criteria.short_name_id:
-            query = query.filter(EFTGLTransferModel.short_name_id == search_criteria.short_name_id)
-
-        if search_criteria.source_gl:
-            query = query.filter(EFTGLTransferModel.source_gl == search_criteria.source_gl)
-
-        if search_criteria.target_gl:
-            query = query.filter(EFTGLTransferModel.target_gl == search_criteria.target_gl)
+        query = query.order_by(EFTGLTransferModel.created_on.asc())
 
-        if search_criteria.transfer_type:
-            query = query.filter(EFTGLTransferModel.transfer_type == search_criteria.transfer_type)
+        return query.all()
 
-        if search_criteria.transfer_date:
-            query = query.filter(func.DATE(EFTGLTransferModel.transfer_date) == search_criteria.transfer_date.date())
+    @staticmethod
+    def filterIfPresent(query, search_criteria, model_attribute):
```

Comment:
This should be generic, put it in the base object?

Reply:
As discussed, extended BaseQuery with CustomQuery class that implements filter_conditionally helper